contrib/pyln-proto/pyln/proto/__init__.py

```diff
@@ -3,7 +3,8 @@
 from .invoice import Invoice
 from .onion import OnionPayload, TlvPayload, LegacyOnionPayload
 from .wire import LightningConnection, LightningServerSocket
-from .__version__ import __version__
+
+__version__ = "0.10.2"
```

Comment:
```suggestion
__version__ = "0.10.3"
```

The actual version on pypi is `0.10.2.post0` that should be greater than 0.10.2?

Reply:
We probably should just make it posts, since we increment the minor versions in lockstep with cln itself

Reply:
just worried about the ability of poetry to resolve no number version, I don't see it in the docs https://python-poetry.org/docs/dependency-specification/ and also in the https://github.com/rustyrussell/lnprototest/pull/34 it cause a recursion problem just trying to resolve the `0.10.2.post0`.

However, I'm not sure what can be the real cause of the recursion here because I was not able to reproduce with other dependencies.

Reply:
The relative dependencies we use inside of this repo doesn't care about version numbers, it just takes the version that is in the repo. But I agree, we should verify that a remote user of the library can resolve things correctly. I'll see if I can port `lnprototest` to use poetry and use a `git`-base dependency so that we always develop against `master`

Reply:
Another thing that I noted is the possibility that can cause the recursion is the pyln-proto dependece unfixed inside the pyln-bolt* package

I nothed that with the following poetry file poetry is able to resolve the dependencies
diff --git a/pyproject.toml b/pyproject.toml
index d23e41a..7927dd0 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -10,11 +10,11 @@ authors = [
```
 [tool.poetry.dependencies]
 python = "^3.8"
 pyln-proto = "^0.10.2"
-pyln-bolt1 = "^1.0.1"
-pyln-bolt2 = "^1.0.2"
-pyln-bolt4 = "^1.0.2"
-pyln-bolt7 = "^1.0.2"
-pyln-testing = "^0.9.3"
+#pyln-bolt1 = "^1.0.1"
+#pyln-bolt2 = "^1.0.2"
+#pyln-bolt4 = "^1.0.2"
+#pyln-bolt7 = "^1.0.2"
+#pyln-testing = "^0.9.3"
 coincurve = "13.0.0"
 python-bitcoinlib = "^0.10.0"
 mypy = '0.790'

```

But if I reintroduce one of the pyln-bolt* packages the recursion happens again.

Reply:
So far no luck, there seems to be a bug in `poetry` failing resolution (might be `git` based dependency + path dependencies in that deps). One easy fix is to either publish the `pyln` packages on pypi, or we checkout the lightning repo as a submodule in `lnprototest`

Reply:
Yes, it is very strange how it resolves the dependencies in poetry, IMHO I think the option to pull it as a submodule can solve different problems for local development and with this solution, we can take time to resolve it as PyPI dependencies.

What do you think?

Reply:
Yeah, we'll have to fix this on the lnprototest side I think